SetFlags.cmake

```diff
@@ -62,6 +62,9 @@ macro(set_flags)
 		set(CMAKE_EXE_LINKER_FLAGS_RELEASE    "${CMAKE_EXE_LINKER_FLAGS_RELEASE}    /LTCG")
 		set(CMAKE_SHARED_LINKER_FLAGS_RELEASE "${CMAKE_SHARED_LINKER_FLAGS_RELEASE} /LTCG")
 		set(CMAKE_MODULE_LINKER_FLAGS_RELEASE "${CMAKE_MODULE_LINKER_FLAGS_RELEASE} /LTCG")
+
+		# Make build use Unicode:
+		add_definitions(-DUNICODE -D_UNICODE)
```

Comment:
I don't quite understan the munging bit. Why would "other systems" be affected by a flag and changes limited to Windows compilation?

Also, what do you mean by "we use UTF-8"? Do you mean the encoding of the source files?

On 19 Sep 2015, at 21:52, "worktycho" <notifications@github.com> wrote:

> In SetFlags.cmake:
> 
> > @@ -62,6 +62,9 @@ macro(set_flags)
> >  		set(CMAKE_EXE_LINKER_FLAGS_RELEASE    "${CMAKE_EXE_LINKER_FLAGS_RELEASE}    /LTCG")
> >  		set(CMAKE_SHARED_LINKER_FLAGS_RELEASE "${CMAKE_SHARED_LINKER_FLAGS_RELEASE} /LTCG")
> >  		set(CMAKE_MODULE_LINKER_FLAGS_RELEASE "${CMAKE_MODULE_LINKER_FLAGS_RELEASE} /LTCG")
> > +
> > +		# Make build use Unicode:
> > +		add_definitions(-DUNICODE -D_UNICODE)
> The big problem is that windows uses UCS-2(ish) whereas we use UTF-8. Now UTF-8 can be munged to windows codepage and back fairly safely. But UCS-2(ish) is really easy for UTF-8 systems to munge.
> 
> —
> Reply to this email directly or view it on GitHub.
>